src/main/java/org/openrewrite/java/spring/boot2/search/EntityIdForRepositoryVisitor.java

```diff
@@ -0,0 +1,358 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.spring.boot2.search;
+
+import org.openrewrite.Tree;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.AnnotationMatcher;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.tree.*;
+import org.openrewrite.marker.Marker;
+import org.openrewrite.marker.SearchResult;
+
+import java.util.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * @author Alex Boyko
+ */
+public class EntityIdForRepositoryVisitor<T> extends JavaIsoVisitor<T> {
+
+    private static final String REPOSITORY = "org.springframework.data.repository.Repository";
+    private static final String ID = "org.springframework.data.annotation.Id";
+
+    private static final AnnotationMatcher ID_ANNOTATION_MATCHER = new AnnotationMatcher("@" + ID, true);
+    public static final String ID_CLASS = "idClass";
+
+    protected boolean considerIdField;
```

Comment:
Just for my understanding: Is this field (and the `createMarker(JavaType)` method) declared as protected since you in your own code base subclass it? If so, it would probably make sense to have a simple test case "using" this, so that it doesn't accidentally get reverted.